components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/AssignUserRolesV2FunctionImpl.java

```diff
@@ -76,10 +77,23 @@ public boolean assignUserRolesV2(JsAuthenticationContext context, List<String> r
 
         List<RoleV2> associatedRoles = Arrays.asList(associatedRolesConfig.getRoles());
         List<RoleV2> allowedRoleListToAssign = new ArrayList<>();
+
         for (String roleName : roleListToAssign) {
-            // Check if the provided role name is associated with the application.
+            // Remove the 'Internal/' prefix if it exists.
+            String processedRoleName = roleName.toLowerCase().startsWith(INTERNAL_DOMAIN_PREFIX)
```

Comment:
check the possibility of using UserCoreUtil.extractDomainFromName here to derive the domain

Reply:
User UserCoreUtil to remove internal domain